docs/tutorial.rst

```diff
@@ -1385,6 +1361,37 @@ may have one of the following signatures:
         # examples the return code would be 0/success.
         return (None, "I failed", 2)
 
+        # Anything you print to stdout or stderr
+        # will be captured for you automatically. This allows callable
+        # aliases to support piping.
+        print('I go to stdout and will be printed or piped')
+
+        # Note: that you have access to the xonsh
+        # built-ins if you 'import builtins'.  For example, if you need the
+        # environment, you could do the following:
+        import builtins
+        env = builtins.__xonsh__.env
+
+    def mycmd1(args):
+        """This form takes a single argument, args. This is a list of strings
+        representing the arguments to this command. Feel free to parse them
+        however you wish!
+        """
+        # perform some action.
+        print(f"arg count: {len(args)}")
+        return 0
+
+    def mycmd2(args, stdin=None):
+        """This form takes two arguments. The args list like above, as a well
+        as standard input. stdin will be a file like object that the command
+        can read from, if the user piped input to this command. If no input
+        was provided this will be None.
+        """
+        # Read input either from piped input or the terminal
+        stdin = stdin or sys.stdin
```

Comment:
@anki-code This line was the core reason for this PR, so we should discuss it.

I found the distinction between stdin and sys.stdin surprising to the point of being a bug. These are the same for normal unix processes. I would have preferred this line being the default behavior, with them only being distinguished using the `spec` argument. However, the preference seemed to be for documenting the behavior rather than changing it.

I'm not sure how to be clearer in this example that stdin comes from piped input and sys.stdin comes from the terminal. To me the comment seems clear, and for most uses the distinction doesn't matter. Can you clarify what you would improve the example?

Reply:
Why would you ever want the alias to work only with piped input but not typed input? I'm all for "real" examples but I think in real cases you should always include the `stdin or sys.stdin` line. This is how all unix tools work (cat, sed, grep, awk, ...).

On the other hand, I like the use of `@aliases.register`. Should all these examples be converted to use the decorator?

Reply:
Is a change required before merging this? I prefer it as is.